README
NAME
    TestML - A Generic Software Testing Meta Language

SYNOPSIS
        # file t/testml/encode.tml
        %TestML 0.1.0

        Title = 'Tests for AcmeEncode'
        Plan = 3

        *text.apply_rot13 == *rot13
        *text.apply_md5   == *md5

        === Encode some poetry
        --- text
        There once was a fellow named Ingy,
        Combining languages twas his Thingy.
        --- rot13
        Gurer bapr jnf n sryybj anzrq Vatl,
        Pbzovavat ynathntrf gjnf uvf Guvatl.
        --- md5: 7a1538ff9fc8edf8ea55d02d0b0658be

        === Encode a password
        --- text: soopersekrit
        --- md5: 64002c26dcc62c1d6d0f1cb908de1435

    This TestML document defines 2 assertions, and defines 2 data blocks.
    The first block has 3 data points, but the second one has only 2.
    Therefore the rot13 assertion applies only to the first block, while the
    the md5 assertion applies to both. This results in a total of 3 tests,
    which is specified in the meta Plan statement in the document.

    To run this test you would have a normal test file that looks like this:

        use TestML;
        use t::Bridge;

        TestML->new(
            testml => 'testml/encode.tml',
            bridge => 't::Bridge',
        )->run;

    The apply_* functions are defined in the bridge class that is specified
    outside this test ("t/Bridge.pm").

DESCRIPTION
    TestML is a generic, programming language agnostic, meta language for
    writing unit tests. The idea is that you can use the same test files in
    multiple implementations of a given programming idea. Then you can be
    more certain that your application written in, say, Python matches your
    Perl implementation.

    In a nutshell you write a bunch of data tests that have inputs and
    expected results. Using a simple syntax, you specify what functions the
    data must pass through to produce the expected results. You use a bridge
    class to write the data functions that pass the data through your
    application.

    In Perl 5, TestML is the evolution of the Test::Base module. It has a
    superset of Test:Base's goals. The data markup syntax is currently
    exactly the same as Test::Base.

SEE ALSO
    *   <http://www.testml.org/>

    *   <irc://irc.freenode.net#testml>

AUTHOR
    Ingy döt Net <ingy@cpan.org>

COPYRIGHT AND LICENSE
    Copyright 2009-2014. Ingy döt Net.

    This program is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself.

    See <http://www.perl.com/perl/misc/Artistic.html>



Changes
0.52 Sat Dec 27 14:45:50 PST 2014
 - Fix file path creation issues on MSWin32

0.51 Sun Dec 14 12:53:02 PST 2014
 - Fix a bug causing 000compile to fail

0.50 Fri Dec 12 10:52:23 PST 2014
 - Make TestML::Setup more clear
 - Share TestML grammar

0.49 Sat Aug 16 12:46:47 PDT 2014
 - Meta 0.0.2

0.48 Sat Aug 16 03:03:29 PDT 2014
 - Eliminate File::Basename from test/

0.47 Sat Aug 16 00:35:15 PDT 2014
 - Move Module::Install::TestML doc to its dist

0.46 Sat Aug 16 00:23:39 PDT 2014
 - Move Module::Install::TestML out of dist

0.45 Fri Aug 15 20:51:09 PDT 2014
 - Add t/000-compile-modules.t

0.44 Wed Aug 13 16:27:24 PDT 2014
 - Fixd issue/4, proper ONLY handling

0.43 Sat Aug  9 00:25:05 PDT 2014
 - Fix swim errors

0.42 Mon Aug  4 12:37:49 PDT 2014
 - Support === and --- as text

0.41 Mon Aug  4 00:59:19 PDT 2014
 - Finish doc to Swim

0.40 Mon Aug  4 00:42:55 PDT 2014
 - Remove (c) from Copyright

0.39 Sat Aug  2 23:55:37 PDT 2014
 - Fix years in Meta

0.38 Sat Aug  2 23:50:11 PDT 2014
 - Fix copyright years.

0.37 Wed Jul 16 23:44:36 PDT 2014
 - Move doc to Swim
 - Fix Meta and add Contributing file

0.36 Sun Jun 22 16:22:12 PDT 2014
 - Use manual VERSION

0.35 Sun Jun 15 13:01:01 PDT 2014
 - Add Chomp to Standard Library

0.34 Sat Jun 14 12:53:08 PDT 2014
 - New Meta updates

0.33 Sun Jun  8 12:31:30 PDT 2014
 - Add badges

0.32 Sat Jun  7 10:05:42 PDT 2014
 - zild release
 - Switch to PkgVersion

0.31 Sun May 18 01:08:48 PDT 2014
 - Upgrade to Pegex 0.30
 - Change to new release setup

0.30 Mon Feb 10 15:09:39 PST 2014
 - Make sure Capture::Tiny is available for testing.

0.29 Sat Feb  8 19:22:48 PST 2014
 - Switch to dzil

0.28 Thu Nov  8 10:49:04 PST 2012
 - Require Pegex-0.21 API changes

0.27 Thu Aug 30 00:07:58 PDT 2012
 - Use new Pegex version

0.26 Mon Nov  7 20:27:00 PST 2011
 - Module::Install::TestML includes Pegex::Input.

0.25 Sat Nov  5 01:11:22 PDT 2011
 - Don't actually use Pegex.pm to get it in the META.

0.24 Fri Sep 30 00:47:16 CEST 2011
 - Use new Mo, inlined.

0.23 Wed Sep 21 23:47:33 CEST 2011
 - Totally refactored Pegex grammar.
 - Released from Whoopie's room in Liz++ and Wendy++'s house.

0.22 Fri Aug 19 20:54:29 EEST 2011
 - Ship with new Pegex.

0.21 Fri Aug 19 20:53:43 EEST 2011
 - Make TestML::Setup work better.

0.20 Thu Sep  9 10:47:15 PDT 2010
 - Total language rewrite.
 - Added functions definitions and calls.
 - Added scoped variables.

0.11 Mon Aug 16 11:49:12 PDT 2010
 - Had a small problem with the transmission...
 - TestML files didn't get into the MANIFEST because of symlink stuffs.
   Fixed.

0.10 Sat Aug 14 17:47:26 PDT 2010
 - Add a light type system to context variables.

0.08 Tue Aug 10 17:24:07 PDT 2010
 - Fixed bug so OK(); can be just OK;

0.07 Tue Aug 10 14:49:21 PDT 2010
 - The Boston.pm release!! patch++
 - Add HAS/OK support.
 - Add Not support.
 - Start testing truth.

0.06 Tue Aug 10 01:56:55 PDT 2010
 - Update Module::Install::TestML
 - Remove TestML::CLI and bin/testml

0.05 Tue Aug 10 01:45:13 PDT 2010
 - Uses Pegex (moved from Parse::Pegex)

0.04 Tue Aug  3 02:50:49 PDT 2010
 - Major update
 - Perl6 implementation backports
 - New grammar spec
 - Uses Parse::Pegex

0.03 Sun May 24 09:42:06 PDT 2009
 - Add Catch support
 - Tweaks and bug fixes
 - Rewrite the runtime and change to the spec requiring $point syntax.

0.02 Thu May 21 19:27:50 PDT 2009
 - Complete rewrite.
 - Written to spec. http://testml.org/spec/

0.01 Thu Apr 23 05:43:45 PDT 2009
 - Initial version.



Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.029.
use strict;
use warnings;

use 5.008001;

use ExtUtils::MakeMaker;

use File::ShareDir::Install;
$File::ShareDir::Install::INCLUDE_DOTFILES = 1;
$File::ShareDir::Install::INCLUDE_DOTDIRS = 1;
install_share dist => "share";


my %WriteMakefileArgs = (
  "ABSTRACT" => "A Generic Software Testing Meta Language",
  "AUTHOR" => "Ingy d\x{f6}t Net <ingy\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::ShareDir::Install" => "0.06"
  },
  "DISTNAME" => "TestML",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.008001",
  "NAME" => "TestML",
  "PREREQ_PM" => {
    "Pegex" => "0.30"
  },
  "TEST_REQUIRES" => {
    "Capture::Tiny" => 0
  },
  "VERSION" => "0.52",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Capture::Tiny" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::ShareDir::Install" => "0.06",
  "Pegex" => "0.30"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}


